imap_processing/ultra/l1b/ultra_l1b_annotated.py

```diff
@@ -34,21 +33,31 @@ def get_particle_velocity(
 
     Returns
     -------
-    spacecraft_velocity : np.ndarray
+    particle_velocity_spacecraft : np.ndarray
         Particle velocity in the spacecraft frame.
+    particle_velocity_heliosphere : np.ndarray
+        Particle velocity in the heliosphere frame.
 
     References
     ----------
     https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy
     """
     # Particle velocity in the pointing (DPS) frame wrt spacecraft.
-    spacecraft_velocity = np.zeros((len(time), 3), dtype=np.float64)
+    particle_velocity_spacecraft = frame_transform(
+        et=time,
+        position=instrument_velocity,
+        from_frame=instrument_frame,
+        to_frame=pointing_frame,
+    )
 
-    for index in range(len(time)):
-        # Get and apply the rotation matrix to the particle velocity.
-        spacecraft_velocity[index] = spice.mxv(
-            spice.pxform(instrument_frame.name, pointing_frame.name, time[index]),
-            instrument_velocity[index],
-        )
+    # Spacecraft velocity in the pointing (DPS) frame wrt heliosphere.
+    state, lt = spice.spkezr("IMAP", time, "IMAP_DPS", "NONE", "SUN")
```

Comment:
There is a `geometry.py` function for this: https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/dev/imap_processing/spice/geometry.py#L85

Pretty much anywhere you directly use spice, it would be good to ask yourself "Would this functionality be useful to other instruments?" If so, the implementation should be moved to `geometry.py` so that others can reuse your code.